### ISSUE TITLE
 Deprecate <none> element in Core

### ISSUE THREAD

**Issue by dginev** [w3c/mathml-core#173]
Hi @fred-wang !

In the MathML Full meeting today (November 10, 2022), the group had consensus that we should explore further a deprecation for the `<none>` element, based on my [issue #428](https://github.com/w3c/mathml/issues/428) in the MathML issues.

I was tasked by the group to also open a mirroring issue in the MathML Core repository, and ask you (Frédéric) if the existing implementations of MathML Core are in full compliance with the text describing `<none>` in the Core spec, or if there are any special implementation details we should be aware of.

If no further changes are needed, we are likely to proceed with a deprecation step. Does that sound good on your end? Thanks!

**Comment by fred-wang** [comment]
Yes, `<none>` does not seem to be used anywhere in WebKit, Chromium, Firefox for layout or accessibility. Its only use in Firefox seems to be to display a special warning message when used outside in mub/msup/msubsup.

I believe it can just be removed from MathML Core (and so treated as unknown element) so it continues to be laid out as an `<mrow>` when it has `display: math`.

For browsers, this will also allow to get rid of automatically generated code for declared tags.

**Comment by dginev** [comment]
Got it, thank you for the fast reply! 

We can continue the discussion in the meeting next Thursday and move things along.

**Comment by fred-wang** [comment]
So actually in Firefox, a `<none/>` child of msub/msup/msubsup, or as the first child of the mmultiscripts, was causing the element to be rendered as "invalid". This special handling is removed in https://bugzilla.mozilla.org/show_bug.cgi?id=1802336

If `<none>` is removed from MathML Core, as I suggested above, I understand this means validator would not allow that element in MathML markup. So I guess for now we can just recommend to use and empty `<mrow>` instead. Note that MathML Core does not use the term "deprecated". Elements/attributes are just not included, or mentioned without defining "any observable behavior" ; and there are non-normative notes explaining this is for backward compatibility.

I'll upload a PR to propose a change: https://github.com/w3c/mathml-core/pull/177

**Comment by dginev** [comment]
> If <none> is removed from MathML Core, as I suggested above, I understand this means validator would not allow that element in MathML markup. So I guess for now we can just recommend to use and empty <mrow> instead.

Indeed, removal leads to `<none/>` being unknown in HTML. And then getting the unknown treatment of falling back to `<mrow/>`. Which is what makes the removal so tempting.

I believe the group discussion last week was explicitly OK with an immediate removal from Core, for what that's worth. The consensus was to mark the deprecation in MathML 4 Full (+removal in MathML 5 if nothing changes), together with a simple removal in MathML Core. Everyone liked the benefit of some harmless simplifying of the Core subset, just before it hits the CR milestone.

Now, I must say that I am personally still learning the fine differences in the processes here, and I was originally expecting a slower timeline than the one currently unfolding. So I feel obliged to tag @davidcarlisle to clarify any misstatements I may have made in my summary here. David has taken up the action item to mark the deprecation in the Full spec.

**Comment by NSoiffer** [comment]
@dginev: even once something moves to CR, it can be removed. Indeed, if some feature doesn't get two implementations, it has to be removed. There's nothing magical about CR in that sense.

**Comment by fred-wang** [comment]
To reiterate, for browser implementations, removing `<none/>` from MathML Core is totally fine, I'm happy to take that way, if that's not a problem for non-browser implementations.

**Comment by NSoiffer** [comment]
Core is focused on browsers (definitions use CSS). Also, I think only browsers have implemented core so far, so nothing breaks.

There is not even a need for a polyfill since `<none/>` is treated like `<mrow/>`.

**Comment by fred-wang** [comment]
Alternative PR, removing none completely: https://github.com/w3c/mathml-core/pull/178

The Firefox's commit has landed and tests merged in WPT already: https://github.com/web-platform-tests/wpt/pull/37173

**Comment by fred-wang** [comment]
`<none>` is listed on MathML AAM too, we likely want to remove it and provide a default mapping for unknown MathML Core elements: https://w3c.github.io/mathml-aam

**Comment by fred-wang** [comment]
We'll also need to update MDN pages e.g. https://developer.mozilla.org/en-US/docs/Web/MathML/Element/mmultiscripts
